File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.cs
public class Solution {
    public int MinOperations(int[] nums, int x) {
        int sz = nums.Length;
int rem = nums.Sum();
rem -= x;

if (rem == 0) return sz;
if (rem < 0) return -1;

IDictionary<int, int> map = new Dictionary<int, int>();
map.Add(0, -1);

int curSum = 0, res = int.MinValue;
for (int i = 0; i < sz; i++)
{
    curSum += nums[i];

    if (map.ContainsKey(curSum - rem))
        res = Math.Max(res, i - map[curSum - rem]);

    map.Add(curSum, i);
}

return res == int.MinValue ? -1 : sz - res;
    }
}

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        long long tot = accumulate(nums.begin(), nums.end(), 0);
int sz = nums.size();
long long rem = tot - x;
int maxWindow = 0, curWindow = 0, curSum = 0;

if (tot == x) return sz;
if (tot < x) return -1;

curSum = nums[0];
for (int i = 0, j = 1; j < sz; ) {
    if (curSum == rem)
        maxWindow = max(maxWindow, j - i);

    if (curSum <= rem)
        curSum += nums[j++];
    while (curSum > rem && i < j)
        curSum -= nums[i++];
    if (curSum == rem)
        maxWindow = max(maxWindow, j - i);
    //cout << curSum << endl;
}

return maxWindow == 0 ? -1 : sz - maxWindow;
    }
};